### ISSUE TITLE
state with store not working

### ISSUE THREAD

**Issue by yathink3** [teafuljs/teaful#36]
```
import { useState } from "react";
import createStore from "teaful";

const initialState = {
  username: "Aral",
  age: 31,
  cart: {
    price: 0,
    items: []
  }
};

export const { useStore, getStore, withStore } = createStore(initialState);

export default function CartFirstItem() {
  const [state,setState] = useState(0);
  const [item, setItem, resetItem] = useStore.cart.items[state]();

  console.log("Render CartFirstItem", item);

  return (
    <>
      <h2>Item: {JSON.stringify(item)}</h2>
      <button
        onClick={async() =>{
          await setItem({ name: "new Item", price: (item?.price || 0) + 1 })
        }}
      >
        Update item
      </button>
      <button onClick={()=>setState(2)}>change Index</button>
      <button onClick={resetItem}>Reset item</button>
    </>
  );
}
```
on this example state index is changed to 2 , but still it not taken 2 index data

**Comment by aralroca** [comment]
Thank you for reporting it @yathink3. I will try to reproduce and correct it today.

**Comment by aralroca** [comment]
@yathink3 fixed on [0.7.2](https://github.com/teafuljs/teaful/releases/tag/0.7.2) 😊 Thank you for reporting it!👏

**Comment by yathink3** [comment]
welcome, i liked this library than any other state management solution for react , 
its dynamic fetching is freaking awesome
